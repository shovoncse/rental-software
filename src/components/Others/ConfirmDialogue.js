import React from 'react';
const ConfirmDialogue = ({ selected }) => {
  return (
    <>
      <p className="text-xl">
        Your Estimated Price is{' '}
        <span className="font-bold text-blue-600">${selected.estPrice}</span>
      </p>
      <p className="text-lg font-semibold">Do you want to proceed ?</p>
    </>
  );
};

export default ConfirmDialogue;
