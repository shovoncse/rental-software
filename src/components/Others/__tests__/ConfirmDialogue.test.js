import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import ConfirmDialogue from '../ConfirmDialogue';
afterEach(cleanup);

describe('<ConfirmDialogue />', () => {
  const selected = { estPrice: '' };
  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <ConfirmDialogue selected={selected}></ConfirmDialogue>,
      div
    );
  });

  it('should render an <p> tag', () => {
    const {
      container: { firstChild },
    } = render(<ConfirmDialogue selected={selected} />);
    expect(firstChild.tagName).toEqual('P');
  });

  it('should have a class attribute', () => {
    const {
      container: { firstChild },
    } = render(<ConfirmDialogue selected={selected} />);
    expect(firstChild.hasAttribute('class')).toBe(true);
  });

  it('matches snapshot', () => {
    const tree = renderer
      .create(<ConfirmDialogue selected={selected}></ConfirmDialogue>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
