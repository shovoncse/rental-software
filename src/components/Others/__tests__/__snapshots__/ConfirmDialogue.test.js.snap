// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmDialogue /> matches snapshot 1`] = `
Array [
  <p
    className="text-xl"
  >
    Your Estimated Price is
     
    <span
      className="font-bold text-blue-600"
    >
      $
      
    </span>
  </p>,
  <p
    className="text-lg font-semibold"
  >
    Do you want to proceed ?
  </p>,
]
`;
